src/manifests_workflow/component_opensearch.py

```diff
@@ -16,13 +17,13 @@ class ComponentOpenSearch(Component):
     @classmethod
     def checkout(
         self,
-        name,
-        path,
-        opensearch_version,
-        branch="main",
-        snapshot=False,
-        working_directory=None,
-    ):
+        name: str,
+        path: str,
+        opensearch_version: str,
+        branch: str = "main",
+        snapshot: bool = False,
+        working_directory: str = None,
+    ) -> 'ComponentOpenSearch':
```

Comment:
Should this be `ComponentOpenSearch` instead of `'ComponentOpenSearch'`?

Reply:
Good point! Actually no, it will say unresolved reference since I think it must be defined before we could use it in an annotation. The solution could be quote it as a string. 
Ref: https://www.python.org/dev/peps/pep-0484/#forward-references

Reply:
I see, Thanks for explaining.